src/core/lib/channel/metrics.h

```diff
@@ -97,9 +100,14 @@ class GlobalInstrumentsRegistry {
   static void ForEach(
       absl::FunctionRef<void(const GlobalInstrumentDescriptor&)> f);
 
-  static void TestOnlyResetGlobalInstrumentsRegistry();
+ private:
+  friend class GlobalInstrumentsRegistryTestPeer;
```

Comment:
I see, interesting way to separate out test code

Reply:
We use the test peer pattern in a number of places.  If we're adding just one or two simple test-only accessor methods, it seems fine to do it right in the class, but if it's more complex, a test peer is better, because it (a) avoids punching holes in the public API surface that would be used outside of tests and (b) avoids bloating the production code with test-only functionality.